Framework/LiveData/src/LiveDataAlgorithm.cpp

```diff
@@ -178,20 +192,44 @@ bool LiveDataAlgorithm::hasPostProcessing() const {
  *
  * @return ILiveListener_sptr
  */
-ILiveListener_sptr LiveDataAlgorithm::getLiveListener() {
+ILiveListener_sptr LiveDataAlgorithm::getLiveListener(bool start) {
   if (m_listener)
     return m_listener;
 
-  // Not stored? Need to create it
-  std::string inst = this->getPropertyValue("Instrument");
-  m_listener = LiveListenerFactory::Instance().create(inst, true, this);
+  // Create a new listener
+  m_listener = createLiveListener(start);
 
   // Start at the given date/time
-  m_listener->start(this->getStartTime());
+  if (start)
+    m_listener->start(this->getStartTime());
 
   return m_listener;
 }
 
+ILiveListener_sptr LiveDataAlgorithm::createLiveListener(bool connect) {
+  // Get the LiveListenerInfo from Facilities.xml
+  std::string inst_name = this->getPropertyValue("Instrument");
+  std::string conn_name = this->getPropertyValue("Connection");
+
+  const auto &inst = ConfigService::Instance().getInstrument(inst_name);
+  const auto &conn = inst.liveListenerInfo(conn_name);
+
+  // See if listener and/or address override has been specified
+  std::string listener = this->getPropertyValue("Listener");
+  if (listener.empty())
+    listener = conn.listener();
```

Comment:
Thinking over it some more, it might be confusing if we log here.

`listener` (or `address`) being empty here is nothing unusual. It will happen in both of these legitimate calls, for example:

```
>>> StartLiveData(Instrument='ISIS_Histogram', OutputWorkspace='wsOut', UpdateEvery=1, AccumulationMethod='Replace', PeriodList=[1,3], SpectraList=[2,4,6])
>>> StartLiveData(Instrument='ISIS_Histogram', Connection='histo', OutputWorkspace='wsOut', UpdateEvery=1, AccumulationMethod='Replace', PeriodList=[1,3], SpectraList=[2,4,6])
```

Also, `createLiveListener` can be called multiple times over the course of one `StartLiveData` call like that, or during interactive usage of the custom dialog for it. The values of "Listener" and "Address" may change between calls. Not every LiveListener created is actually used. In many places they are created temporarily just to check whether they support events or not, for example, and then discarded immediately.